src/dxvk/rtx_render/rtx_draw_call_cache.cpp

```diff
@@ -85,7 +85,12 @@ DrawCallCache::CacheState DrawCallCache::get(const DrawCallState& drawCall, Blas
 
   float bestScore = std::numeric_limits<float>::min();
   Matrix4 newTransform = drawCall.getTransformData().objectToWorld;
-  const Vector3 newWorldPosition = Vector3(newTransform[3][0], newTransform[3][1], newTransform[3][2]);
+
+  Vector3 newWorldPosition = Vector3(newTransform[3][0], newTransform[3][1], newTransform[3][2]);
+  if (RtxOptions::Get()->instanceUseBoundingBox()) {
+    newWorldPosition = (newTransform * Vector4(drawCall.getGeometryData().boundingBox.getCentroid(), 1.0f)).xyz();
```

Comment:
You follow the pattern of "get position, check RtxOption, replace position with transform * centroid" several times.

You could move all of this logic into the AABB class, and just depend on the whether the AABB is valid for the draw call instead of checking the RtxOption over and over:

```
Vector3 getTransformedCentroid(const Matrix4& transform) {
  if (isValid()) {
    return (transform * Vector4(getCentroid(), 1.f)).xyz();
  } else {
    return transform[3].xyz();
  }
}
```

That way we just default to using the more accurate measurement when it's available.  If calculating the AABB ever shows up as a perf bottleneck in the future, we could easily swap to only calculating AABB's that are in a specific texture list or something, instead of doing it for every draw call.

Reply:
Fair point! I've included a toggle so that the feature can be tested / validated more easily.
Should I squash again or keep it as a second commit?